cmd/globalAlignmentAnchor/globalAlignmentAnchor.go

```diff
@@ -25,40 +25,78 @@ func writeToFileHandle(file io.Writer, species1 bed.Bed, species2 bed.Bed, score
 	exception.PanicOnErr(err)
 }
 
+// helper function: check if maf entry passes checks
+func matchMafPass(assembly_species1 string, assembly_species2 string, chrom_species1 string, chrom_species2 string, species1_SrcSize int, species2_SrcSize int, species1_ChromStart int, species1_ChromEnd int, species2_ChromStart int, species2_ChromEnd int) bool {
+	pass := true
+
+	// chrom should be the same between species1 and species2 (special case: panTro6 chr2A, chr2B both count as a match to hg38 chr2)
+	pT6_hg38_chr2 := ((assembly_species1 == "hg38" && assembly_species2 == "panTro6" && chrom_species1 == "chr2" && ((chrom_species2 == "chr2A") || (chrom_species2 == "chr2B"))) || (assembly_species1 == "panTro6" && assembly_species2 == "hg38" && ((chrom_species1 == "chr2A") || (chrom_species1 == "chr2B")) && chrom_species2 == "chr2")) // the special case written as a bool variable
+	if chrom_species2 != chrom_species1 || pT6_hg38_chr2 {
+		pass = false
+	}
+
+	// maf entry should be roughly diagonal
+	if (float64(species2_ChromStart) <= float64(species1_ChromStart)-0.05*float64(species1_SrcSize)) || (float64(species2_ChromStart) >= float64(species1_ChromStart)+0.05*float64(species1_SrcSize)) {
+		pass = false
+	} else if (float64(species1_ChromStart) <= float64(species2_ChromStart)-0.05*float64(species2_SrcSize)) || (float64(species1_ChromStart) >= float64(species2_ChromStart)+0.05*float64(species2_SrcSize)) {
+		pass = false
+	}
+
+	return pass
+}
+
 // helper function: check if gap bed entry passes checks
-func gapBedPass(species1_ChromStart int, species1_ChromEnd int, species2_ChromStart int, species2_ChromEnd int, gapSizeProductLimit int) (bool, string, string) {
+// need both pos_species (most recently updated alignment position)
+// and species1_ChromStart (the start of the current bed entry)
+func gapBedPass(pos_species1 int, species1_ChromStart int, species1_ChromEnd int, pos_species2 int, species2_ChromStart int, species2_ChromEnd int, gapSizeProductLimit int) (bool, string, string) {
```

Comment:
The old code in this file labels each function as "Step 1..." , "Step 2..." , etc. I liked that it had a flow, but I notice that these new functions are not numbered.

Reply:
Thanks! That's because these functions are helper functions I call in some of the steps, but they are not steps themselves. For example, gapBedPass is called in Step2 matchToGap